engine/src/main/java/org/terasology/rendering/dag/RenderGraph.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import com.google.common.collect.Lists;
+import java.util.List;
+
+/**
+ *
+ */
+public class RenderGraph { // TODO: add extends DirectedAcyclicGraph<Node>
+    private List<Node> nodes;
+
+    public RenderGraph() {
+        nodes = Lists.newArrayList();
+    }
+
+    public String add(Node node, String suggestedId) {
+        nodes.add(node);
+        return null; // TODO: for instance if "blur" present make id "blur1" and return it
+    }
+
+    public String add(Node node) {
```

Comment:
I completely agree about frustration of reading the log: finding out the failure of a randomly labeled node. What I had in mind was if the class of that node is named as `FooNode`, I would expect it in logs as "Node fooNode failed to process", I think we can manage to do that in java by [Class.getSimpleName()](http://docs.oracle.com/javase/6/docs/api/java/lang/Class.html#getSimpleName%28%29).

But for the time being, let me remove that one. We can discuss this in the future.